baremetal_network_provisioning/drivers/discovery_driver.py

```diff
@@ -13,18 +13,46 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from baremetal_network_provisioning.common import constants
+from baremetal_network_provisioning.drivers import snmp_client
 
-class DiscoveryDriver(object):
 
-    """Driver for All family of switches to get the switch and port
+class SNMPDiscoveryDriver(object):
 
-    information.
+    def __init__(self, snmp_info):
+        self.snmp_info = snmp_info
+        self.client = snmp_client.get_client(snmp_info)
 
-    """
+    def discover_switch(self):
+        mac_addr = self.get_mac_addr()
+        ports_dict = self.get_ports_info()
+        switch = {'mac_addr': mac_addr, 'ports': ports_dict}
+        return switch
 
-    def initialize(self):
-        pass
+    def get_mac_addr(self):
+        oid = constants.OID_MAC_ADDRESS
+        varBinds = self.client.get(oid)
```

Comment:
will take up in next sprint. I think Mac address and ports will never be empty/None .